### PR TITLE
[MRG] Adding `channel_wise` argument to `Raw.apply_function`

### PR DESCRIPTION
**Pull request by hubertjb**
See issue #5860.

This adds an argument `channel_wise` to `mne.io.Raw.apply_function()`. If `channel_wise` is True (default), the function is applied on each channel individually; if it is False, then the whole data array is passed to the function at once.

Questions:
1) Should we raise an error or display a warning if `channel_wise` is False and `n_jobs` is not 1?
2) I added a few lines to the existing test `test_apply_function_verbosity`. Would it better to create a separate test function?

### CODE REVIEW

**Comment by larsoner** [comment]
> Should we raise an error or display a warning if channel_wise is False and n_jobs is not 1?

I would just say in the docstring that it's ignored

>  Would it better to create a separate test function?

It's fine to add to an existing test for a small/simple addition like this

File: mne/io/base.py [hunk 1]
```diff
@@ -1066,8 +1066,12 @@ def apply_function(self, fun, picks=None, dtype=None,
         dtype : numpy.dtype (default: None)
             Data type to use for raw data after applying the function. If None
             the data type is not modified.
+        channel_wise: bool (default: True)
+            Whether to apply the function to each channel individually. If
+            False, the function will be applied to all channels at once.
```
Comment (larsoner) [inline mne/io/base.py]: It's probably worth modifying the docstring for `fun` to say that it needs to operate on an array of shape `(n_times,)` if `channel_wise=True` and `(len(picks), n_times)` if `channel_wise=False`

File: mne/io/tests/test_apply_function.py [hunk 2]
```diff
@@ -40,6 +45,11 @@ def test_apply_function_verbose():
     pytest.raises(TypeError, raw.apply_function, bad_1, n_jobs=2)
     pytest.raises(ValueError, raw.apply_function, bad_2, n_jobs=2)
 
+    # test return type when `channel_wise=False`
+    raw.apply_function(printer, channel_wise=False)
+    pytest.raises(TypeError, raw.apply_function, bad_1, channel_wise=False)
```
Comment (larsoner) [inline mne/io/tests/test_apply_function.py]: the "modern" way of writing these is to do:
```
with pytest.raises(TypeError, match='must be blah blah blah'):
    raw.apply_function(bad_1, channel_wise=False)
```
it makes it clearer what's being caught and why, and ensures that's the case

**Comment by hubertjb** [comment]
Thanks @larsoner for the comments. I just updated the PR.

**Comment by larsoner** [comment]
Pushed a tiny change to fix the order (new kwarg last) and RST fixes (double-backticks for code)

**Comment by larsoner** [comment]
Please set to MRG if it's ready to go from your end

+1 for merge from me

**Comment by massich** [comment]
Cool update what's new and I'll merge it! Thx @hubertjb

**Comment by agramfort** [comment]
travis failure is unrelated. Merging.

**Comment by agramfort** [comment]
thx @hubertjb

**Pull request merged**